File: Java/interview_24.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if (head == null) {
            return head;
        }
        ListNode prev = head;
        ListNode latter = head.next;
        head.next = null;
        while (latter != null) {
            ListNode tmp = latter.next;
            latter.next = prev;
            prev = latter;
            latter = tmp;
        }
        return prev;
    }
}

File: python/interview_24.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        if head is None:
            return None
        prev = head
        latter = head.next
        head.next = None
        while latter is not None:
            tmp = latter.next
            latter.next = prev
            prev = latter
            latter = tmp
        return prev
